### PR TITLE
Destination Snowflake: Use safe executeMetadataQuery call

### PR DESCRIPTION
**Pull request by gisripa**
## What

`database.getMetadata()` recycles the underlying connection object back into the pool (in hikari land) after the method call returns. DbMetadata holds an instance copy of that connection object and any subsequent calls which yield a `ResultSet` could potentially deadlock other calls using the same connection.  This was already fixed in `JdbcDestinationHandler` but this call was missed out. 

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by edgao — APPROVED** [review]
ffs

**Comment by edgao** [comment]
doublechecking: resultSet.getMetadata() is safe? b/c presumably we're already inside a database.query() block

(we have a few rs.getmetadata calls in various places)

**Comment by gisripa** [comment]
> doublechecking: resultSet.getMetadata() is safe? b/c presumably we're already inside a database.query() block
> 
> (we have a few rs.getmetadata calls in various places)

Yeah, as long as the rs is in try-with or `.use{}` we are good.  db.getMetadata should be wiped but sources use it for static information like `getQuoteLiteral` etc

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java [hunk 1]
```diff
@@ -132,12 +132,18 @@ private InitialRawTableStatus getInitialRawTableState(final StreamId id, final D
...
-    final ResultSet tables = database.getMetaData().getTables(
-        databaseName,
-        id.getRawNamespace(),
-        id.getRawName(),
-        null);
-    if (!tables.next()) {
+    final boolean tableExists = database.executeMetadataQuery(databaseMetaData -> {
+      LOGGER.info("Retrieving table from Db metadata: {} {}",
+          id.getRawNamespace(),
+          id.getRawName());
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java:138]: 👍 for logging (for now)

**Pull request merged**